panther_battery/test/test_adc_data_reader.cpp

```diff
@@ -0,0 +1,61 @@
+#include <filesystem>
+#include <fstream>
+#include <memory>
+
+#include <gtest/gtest.h>
+
+#include <panther_battery/adc_data_reader.hpp>
+
+class TestADCDataReader : public testing::Test
+{
+public:
+  TestADCDataReader();
+  ~TestADCDataReader();
+
+protected:
+  std::shared_ptr<panther_battery::ADCDataReader> data_reader_;
+  std::filesystem::path current_path_;
+  std::filesystem::path file_path_;
+  std::ofstream file_;
+
+  void WriteNumberToFile(const int & number);
+};
+
+TestADCDataReader::TestADCDataReader()
+{
+  current_path_ = std::filesystem::current_path();
+  file_path_ = current_path_ / "test_file";
```

Comment:
This should not be regular file, but rather [tmpfile](https://en.cppreference.com/w/cpp/io/c/tmpfile), stored in `/tmp` directory